File: LHDS.Core/Models/Foundations/Datasets/Dataset.cs
// ---------------------------------------------------------------
// Copyright (c) North East London ICB. All rights reserved.
// ---------------------------------------------------------------

using System;
using System.Collections.Generic;
using LHDS.Core.Models.Bases;
using LHDS.Core.Models.Foundations.DataSetSpecifications;

namespace LHDS.Core.Models.Foundations.DataSets
{
    public class DataSet : IKey, IAudit
    {
        public Guid Id { get; set; }
        public string DataSetName { get; set; }
        public string DataSetAliasses { get; set; }
        public string DataSetSupplier { get; set; }
        public string DataSetAuthor { get; set; }
        public string SpecifiedBy { get; set; }
        public bool IsNationallySpecified { get; set; }
        public string CollectedBy { get; set; }
        public bool IsNationallyCollected { get; set; }
        public string DataSourceType { get; set; }
        public bool IsActive { get; set; }
        public DateTimeOffset? ActiveFrom { get; set; }
        public DateTimeOffset? ActiveTo { get; set; }
        public string CreatedBy { get; set; }
        public string UpdatedBy { get; set; }
        public DateTimeOffset UpdatedDate { get; set; }
        public DateTimeOffset CreatedDate { get; set; }

        public List<DataSetSpecification> DataSetSpecifications { get; set; } = new List<DataSetSpecification>();
    }
}

File: LHDS.AdminPortal.Web/ClientApp/src/models/dataSets/dataSet.ts
import { Guid } from 'guid-typescript';

export class DataSet {
    public id: Guid;
    public dataSetName?: string;
    public dataSetAliasses?: string;
    public dataSetSupplier?: string;
    public dataSetAuthor?: string;
    public dataSourceType?: string;
    public isActive?: boolean;
    public activeFrom?: Date;
    public activeTo?: Date;
    public createdBy?: string;
    public createdDate?: Date;
    public updatedBy?: string;
    public updatedDate?: Date;

    constructor(dataSet: any) {
        this.id = dataSet.id ? Guid.parse(dataSet.id) : Guid.parse(Guid.EMPTY);
        this.dataSetName = dataSet.dataSetName;
        this.dataSetAliasses = dataSet.dataSetAliasses;
        this.dataSetSupplier = dataSet.dataSetSupplier;
        this.dataSetAuthor = dataSet.dataSetAuthor;
        this.dataSourceType = dataSet.dataSourceType;
        this.isActive = dataSet.isActive;
        this.activeTo = dataSet.activeTo;
        this.createdDate = dataSet.createdDate;
        this.createdBy = dataSet.createdBy;
        this.createdDate = new Date(dataSet.createdDate);
        this.updatedBy = dataSet.updatedBy;
        this.updatedDate = new Date(dataSet.updatedDate);
    }
}